google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -1763,4 +1807,52 @@ public void onSessionCreateFailure(Throwable t, int createFailureForSessionCount
       }
     }
   }
+
+  /**
+   * Initializes and Creates Spanner session relevant metrics. When coupled with an exporter, it
+   * allows users to monitor client behavior.
+   */
+  private void initMetricsCollection(MetricRegistry metricRegistry, List<LabelValue> labelValues) {
+    DerivedLongGauge activeSessionsGauge =
+        metricRegistry.addDerivedLongGauge(
+            ACTIVE_SESSIONS,
+            MetricOptions.builder()
+                .setDescription(ACTIVE_SESSIONS_DESCRIPTION)
+                .setUnit(COUNT)
+                .setLabelKeys(SPANNER_LABEL_KEYS)
+                .build());
+
+    DerivedLongGauge maxSessionsGauge =
+        metricRegistry.addDerivedLongGauge(
+            MAX_SESSIONS,
+            MetricOptions.builder()
+                .setDescription(MAX_SESSIONS_DESCRIPTION)
+                .setUnit(COUNT)
+                .setLabelKeys(SPANNER_LABEL_KEYS)
+                .build());
+
+    // The value of a numSessionsInUse is observed from a callback function. This function is
+    // invoked whenever metrics are collected.
+    activeSessionsGauge.createTimeSeries(
+        labelValues,
+        this,
+        new ToLongFunction<SessionPool>() {
+          @Override
+          public long applyAsLong(SessionPool sessionPool) {
+            return sessionPool.numSessionsInUse;
```

Comment:
I am not 100% sure on `numSessionsInUse`, may be we should use `maxSessionsInUse` to indicate active sessions in use. Please confirm.

Reply:
Agree, maxSessionInUse will be more useful. It resets every 10 minutes while the reporting interval is 1 minute. Even if it was reset every 1 minute it would not be aligned. So it needs to reset on read.